components/raftstore/src/store/peer_storage.rs

```diff
@@ -887,8 +879,19 @@ where
         if self.truncated_term() == self.last_term || idx == self.last_index() {
             return Ok(self.last_term);
         }
-        let entries = self.entries(idx, idx + 1, raft::NO_LIMIT)?;
-        Ok(entries[0].get_term())
+        if let Some(e) = self.cache.entry(idx) {
+            Ok(e.get_term())
+        } else {
+            let mut entries = vec![];
```

Comment:
You mean `self.engines.raft.get_entry(self.get_region_id(), idx)`?

Reply:
Yep. It's the original code. I change to use `get_entry`.